internal/hierarchy/hierarchy.go

```diff
@@ -48,6 +48,22 @@ func (p pathValue) Equal(o any) bool {
 	return true
 }
 
+// GetEndpoint returns the hierarchical path of endpoint.
+func GetEndpoint(endpoint resolver.Endpoint) []string {
+	attrs := endpoint.Attributes
+	if attrs == nil {
+		return nil
+	}
+	path, _ := attrs.Value(pathKey).(pathValue)
+	return ([]string)(path)
+}
+
+// SetEndpoint overrides the hierarchical path in endpoint with path.
+func SetEndpoint(endpoint resolver.Endpoint, path []string) resolver.Endpoint {
```

Comment:
`SetInEndpoint`?  (If you can think of something better, happy to hear it...)

Reply:
SetInEndpoint sounds good to me.

hierarchy.SetInEndpoint(...)